inst/include/dust/rng.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef DUST_RNG_HPP
+#define DUST_RNG_HPP
+
+#include <random>
+#include <dust/xoshiro.hpp>
+#include <dust/distr/binomial.hpp>
+#include <dust/distr/poisson.hpp>
+
+namespace dust {
+
+class RNG {
+public:
+  RNG(dust::Xoshiro generator) : _generator(generator) {}
+
+  double unif_rand() {
+    return _generator.unif_rand();
+  }
+
+  double runif(double min, double max) {
```

Comment:
Can we template `runif` and `rnorm` too, and set the default to double? Will be useful if/when they become floats